app/scripts/modules/amazon/src/serverGroup/configure/wizard/pages/advancedSettings/ServerGroupAdvancedSettingsCommon.tsx

```diff
@@ -255,41 +273,58 @@ export class ServerGroupAdvancedSettingsCommon extends React.Component<IServerGr
         </div>
         <div className="form-group">
           <div className="col-md-5 sm-label-right">
-            <b>Associate Public IP Address</b>
-          </div>
-          <div className="col-md-2 radio">
-            <label>
-              <input
-                type="radio"
-                checked={values.associatePublicIpAddress === true}
-                onChange={() => setFieldValue('associatePublicIpAddress', true)}
-                id="associatePublicIpAddressTrue"
-              />
-              Yes
-            </label>
-          </div>
-          <div className="col-md-2 radio">
-            <label>
-              <input
-                type="radio"
-                checked={values.associatePublicIpAddress === false}
-                onChange={() => setFieldValue('associatePublicIpAddress', false)}
-                id="associatePublicIpAddressFalse"
-              />
-              No
-            </label>
-          </div>
-          <div className="col-md-2 radio">
-            <label>
-              <input
-                type="radio"
-                checked={values.associatePublicIpAddress === null}
-                onChange={() => setFieldValue('associatePublicIpAddress', null)}
-                id="associatePublicIpAddressDefault"
-              />
-              Default
-            </label>
+            <b>{`Associate ${asgSettings?.enableIPv6 ? 'IPv6' : 'Public IPv4'} Address`}</b>
           </div>
+          {asgSettings?.enableIPv6 && (
+            <div className="col-md-6 checkbox">
+              <label>
+                <input
+                  type="checkbox"
+                  checked={values.associatePublicIpAddress === true}
+                  onChange={e => setFieldValue('associatePublicIpAddress', e.target.checked)}
```

Comment:
associatePublicIpAddress controls whether or not to assign a public ipv4 address on both a launch config and template. 
I need to look at how to auto assign IPv6 with launch templates.

Reply:
When launch templates are enabled, CNE wants assigning a public IP address to be IPv6 only. Any dual-stacked instance will have a private IPv4 and an IPv6. We can either re-use this flag, and perform a different operation in clouddriver when using launch templates or we can use a new flag `assignIPv6`.  Let me know what makes sense. The change on my end will be pretty nominal.

Reply:
Let’s reuse the same flag.